pkg/image/controller/controller.go

```diff
@@ -55,9 +217,6 @@ const retryCount = 2
 // 4. ImageStreamMapping save error
 // 5. error when marking ImageStream as imported
 func (c *ImportController) Next(stream *api.ImageStream) error {
```

Comment:
"barring insanity" is a phrase that is starting to have so little currency
I dread its appearance.  You cannot bar insanity - its eldritch tentacles
crowd every gate and enter every chink in the armor of reality.

On Sun, Dec 20, 2015 at 8:05 AM, David Eads <notifications@github.com>
wrote:

> In pkg/image/controller/controller.go
> <https://github.com/openshift/origin/pull/6407#discussion_r48103161>:
>
> > @@ -55,9 +217,6 @@ const retryCount = 2
> >  // 4. ImageStreamMapping save error
> >  // 5. error when marking ImageStream as imported
> >  func (c *ImportController) Next(stream *api.ImageStream) error {
>
> Hmm, it just occurred to me that we need to make sure this function is
> thread safe. Initializing the shared client here isn't, and I need to sweep
> getTags/importTags and the client functions to see if it is building
> connection caches in a thread safe way.
>
> I agree the thread safety is important, but since client is only "injected
> for testing" per the godoc (and verified in the code), that client looks
> local to me. So it results in one per call, so the connection establishment
> below should be safe, barring insanity.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6407/files#r48103161>.
>